Starting a docker image app fails if it tries to establish connections before Envoy is ready

### Summary

Starting or restarting a docker image app fails if the app tries to establish connections before the Envoy proxy is ready, and this communication failure causes the app to crash.

```
$ cf restart service-manager
Restarting app service-manager in org SAP_servicemanager / space service-manager as admin...

Stopping app...

Waiting for app to start...
Start unsuccessful
```

Examining the corresponding pod shows that the `opi` container was restarted once due to an error. The logs from the first (failed) run are as follows:

```
{"component_type":"application","correlation_id":"-","level":"info","logger":"sm/sm.go:105","msg":"Setting up Service Manager storage...","type":"log","written_at":"2020-05-08T09:23:04.068924213Z","written_ts":"1588929784"}
panic: unable to construct service-manager builder: error opening storage: error opening storage: could not update database schema: dial tcp 100.65.158.99:5432: connect: connection refused

goroutine 1 [running]:
main.main()
	/go/src/github.wdf.sap.corp/SvcManager/sm-sap/main.go:34 +0x783
```

The above logs indicate that the app tried to connect to its database, but the connection was refused, which caused the app to crash. The reason that the connection was refused is that by that time (2020-05-08T09:23:04), the Envoy proxy hasn't started yet, since the `istio-proxy` logs indicate that this happens a second later:

```
2020-05-08T09:23:05.767841Z	info	Envoy proxy is ready
```

After the `opi` container is restarted by Kubernetes, the app starts successfully and connects to its database. So the end result is a running app, but a failed `cf start`.

This is reproducible with several apps on our side. All of them try to establish connections before Envoy has started. Some of them treat the "connection refused" error as fatal and crash, leading to the above error. Others treat it as non-fatal or retry, and so avoid the error.

### Reproduction Steps

Push an app from a docker image that is fairly quick to start, tries to establish a connection immediately after starting, and treats the failure as fatal. For best effect, push it with `--no-start` and then start it with `cf start`, or restart it with `cf restart`.

### Expected behavior

`cf start` or `cf restart` don't return an error in this case. This could be achieved by either ensuring that Envoy proxy starts before the app tries to establish a connection, or treating the first crash from the app as non-fatal.

--- a/build/istio/istioctl-values.yaml
+++ b/build/istio/istioctl-values.yaml
@@ -79,4 +79,6 @@ spec:
         }
     global:
       controlPlaneSecurityEnabled: true
+      proxy:
+        holdApplicationUntilProxyStarts: true
 

--- a/config/istio/istio-generated/xxx-generated-istio.yaml
+++ b/config/istio/istio-generated/xxx-generated-istio.yaml
@@ -8300,7 +8300,7 @@ data:
           "excludeIPRanges": "",
           "excludeInboundPorts": "",
           "excludeOutboundPorts": "",
-          "holdApplicationUntilProxyStarts": false,
+          "holdApplicationUntilProxyStarts": true,
           "image": "proxyv2",
           "includeIPRanges": "*",
           "logLevel": "warning",

